packages/core/src/utils/spanUtils.ts

```diff
@@ -169,15 +169,24 @@ export function getStatusMessage(status: SpanStatus | undefined): string | undef
 }
 
 const CHILD_SPANS_FIELD = '_sentryChildSpans';
+const ROOT_SPAN_FIELD = '_sentryRootSpan';
 
 type SpanWithPotentialChildren = Span & {
   [CHILD_SPANS_FIELD]?: Set<Span>;
+  [ROOT_SPAN_FIELD]?: Span;
 };
 
 /**
  * Adds an opaque child span reference to a span.
  */
 export function addChildSpanToSpan(span: SpanWithPotentialChildren, childSpan: Span): void {
+  // We store the root span reference on the child span
+  // We need this for `getRootSpan()` to work
+  const rootSpan = span[ROOT_SPAN_FIELD] || span;
+  (childSpan as SpanWithPotentialChildren)[ROOT_SPAN_FIELD] = rootSpan;
```

Comment:
l: Should we make this non-enumerable so that it doesn't leak API?

Reply:
yes we should, actually! good point!

Reply:
actually I'll also do this for the span children (which is where I copied this from xD), same applied there I'd say :)